src/components/NotificationProvider/NotificationProvider.tsx

```diff
@@ -0,0 +1,118 @@
+import React, {
+  createContext,
+  FC,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from "react";
+import {
+  NotificationType,
+  NotificationHelper,
+  QueuedNotification,
+  NotifyProviderProps,
+} from "./types";
+import { useLocation } from "react-router-dom";
+import isEqual from "lodash/isEqual";
+import Notification from "../Notification";
+import { info, failure, success, queue } from "./messageBuilder";
+
+const NotifyContext = createContext<NotificationHelper>({
+  notification: null,
+  clear: () => undefined,
+  failure: () => undefined,
+  success: () => undefined,
+  info: () => undefined,
+  queue: () => undefined,
+});
+
+export const NotificationProvider: FC<NotifyProviderProps> = ({ children }) => {
+  const [notification, setNotification] = useState<NotificationType | null>(
+    null
+  );
+  const { state, pathname } = useLocation() as QueuedNotification;
+
+  const clear = () => notification !== null && setNotification(null);
+
+  const setDeduplicated = (value: NotificationType) => {
+    if (!isEqual(value, notification)) {
+      setNotification(value);
+    }
+    return value;
+  };
+
+  useEffect(() => {
+    if (state?.queuedNotification) {
+      setDeduplicated(state.queuedNotification);
+      window.history.replaceState({}, "");
+    } else {
+      clear();
+    }
+  }, [state, pathname]);
+
+  const helper: NotificationHelper = {
+    notification,
+    clear,
+    queue,
+    failure: (title, error, message, actions) =>
+      setDeduplicated(failure(title, error, message, actions)),
+    info: (message, title) => setDeduplicated(info(message, title)),
+    success: (message) => setDeduplicated(success(message)),
+  };
+
+  return (
+    <NotifyContext.Provider value={helper}>{children}</NotifyContext.Provider>
+  );
+};
+
+export function useNotify() {
+  return useContext(NotifyContext);
+}
+
+export const NotificationConsumer: FC = () => {
```

Comment:
I'm not convinced by the way this was integrated in the react-components code base. I think the hook should be separated, and placed in a dedicated file inside the "hooks" folder. The FC only should be nested under components instead.